src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/CreateNSTreeModal/index.tsx

```diff
@@ -100,18 +104,44 @@ export const CreateNSTreeModal = ({
   // Filter based on location
   const { data: groupInfo } = useGroupInfo();
   const { data: namedLocationsData } = useNamedLocations();
-  const namedLocations: NamedGisaidLocation[] =
-    namedLocationsData?.namedLocations ?? [];
+  const namedLocations: NamedGisaidLocation[] = useMemo(() => {
+    return namedLocationsData?.namedLocations ?? [];
+  }, [namedLocationsData]);
 
   // If we have the group's location, use this as the default for the filter
   const [selectedLocation, setSelectedLocation] =
     useState<NamedGisaidLocation | null>(getLocationFromGroup(groupInfo));
 
+  const locationMaxDepthFinder = useMemo(() => {
+    return createMaxDepthLocationFinder(namedLocations);
+  }, [namedLocations]);
+
+  const setLocationToGroupDefault = () => {
+    const locationDepth = locationDepthPathogenConfig[pathogen];
+
+    const defaultTreeLocation =
```

Comment:
i think this might be more readable as actual if/else-if statements, but I hold that lightly as it's personal preference.

Reply:
I like them, but I'll be overruled here.